zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java

```diff
@@ -1,98 +1,202 @@
+import static com.github.onsdigital.logging.v2.event.SimpleEvent.info;
+import static com.github.onsdigital.zebedee.util.URIUtils.removeProtocolHostAndPort;
 
 /**
  * Created by bren on 16/12/15.
  * Publish notification to be sent to the website for caching timing purposes
  */
+
 public class PublishNotification {
 
+    private NotificationPayload payload;
+    private final List<NotificationPayloadCacheApi> payloads = new ArrayList<>();
+    private final String collectionId;
+
     private static final List<Host> websiteHosts;
     private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-    private NotificationPayload payload;
+    private static final String LEGACY_CACHE_API_PATH = "/v1/cache-times/";
+
+    // Url prefix variables
+    private static final String ONS_PREFIX = "/ons/";
+
+    // Segment input variables
+    private static final String ADHOC_SEGMENT = "/adhocs";
+    private static final String ARTICLE_SEGMENT = "/articles";
+    private static final String BULLETIN_SEGMENT = "/bulletins";
+    private static final String DATA_SEGMENT_WITHOUT_SLASH = "data";
+    private static final String DATASETS_SEGMENT = "/datasets";
+    private static final String LINE_CHART_CONFIG_SEGMENT = "/linechartconfig";
+    private static final String QMIS_SEGMENT = "/qmis";
+    private static final String TIME_SERIES_SEGMENT = "/timeseries";
+    private static final String VISUALISATIONS_SEGMENT = "/visualisations";
+
+    // Resource types
+    private static final List<String> RESOURCE_TYPE_SEGMENTS = Arrays.asList(
+            "/chartconfig?",
+            "/chartimage?",
+            "/embed?",
+            "/chart?",
+            "/resource?",
+            "/generator?",
+            "/file?",
+            "/export?"
+    );
+
+    private Http http;
 
     static {
         websiteHosts = Configuration.getWebsiteHosts();
     }
 
     public PublishNotification(Collection collection, List<String> urisToUpdate, List<ContentDetail> urisToDelete) {
+        this.collectionId = collection.getDescription().getId();
 
         // Delay the clearing of the cache after publish to minimise load on the server while publishing.
         Date clearCacheDate = new DateTime(collection.getDescription().getPublishDate())
                 .plusSeconds(Configuration.getSecondsToCacheAfterScheduledPublish()).toDate();
 
-        this.payload = new NotificationPayload(collection.getDescription().getId(), urisToUpdate, urisToDelete,
-                clearCacheDate);
+        if (Configuration.isLegacyCacheAPIEnabled()) {
+            if (urisToUpdate != null) {
+                for (String uriToUpdate : urisToUpdate) {
+                    NotificationPayloadCacheApi notificationPayloadCacheApi = new NotificationPayloadCacheApi(collection.getDescription().getId(), uriToUpdate, clearCacheDate);
+                    payloads.add(notificationPayloadCacheApi);
```

Comment:
Unless this is handled somewhere else, and I didn't notice, I think an important part is missing: the Bulletin /latest pages are not being processed as they should. Please read that section in the diagram to see what I mean.

Reply:
Jake, can you pick up Jos comments that are related to path updates?